contents/docs/contributing.md

```diff
@@ -29,9 +29,22 @@ If you're planning to work on a bigger feature, and it's not in the list of issu
 
 ## Testing
 
+### Backend
 We expect all backend code to have corresponding tests. You can run the tests by running `bin/tests`. To specify a specific test (rather than running all tests) you can run `bin/tests posthog.api.test.test_action` for example.
 
-At the moment, we do not have many frontend tests. Do make sure that each page you've touched (or that uses a component you touched) still functions as expected.
+### Frontend
+We use [cypress](https://www.cypress.io/) for functional frontend tests. Cypress is used to mock user interface interactions on a demo instance. We've currently targeted test coverage for the most used areas of posthog and are working to increase coverage. 
+
+We follow the given, when, then framework. Each test should provide a given (initial setup for the flow or feature being tested), when (an action performed), and then (an assertion that verifies resulting behavior). To keep our components flexible and easily maintainable within tests, we use a `dataattr` pattern where we add a page-unique identifier prop to react components for testing purposes. Then, to identify an element in a cypress tests we use `cy.get('[dataattr=some-identifier]')...`. 
+
+#### Cypress usage for local development
+
+1. Ensure that you either `yarn install` or `yarn add cypress --dev` in the root directory
```

Comment:
Should just be `yarn install` once your PR goes through right?

Reply:
Yup, I can remove the second command. Wrote it in because technically that's all you need to start up the cypress client